### PR TITLE
C#: ChannelOption improvements

### PR DESCRIPTION
**Pull request by jtattermusch**
- Improve doc comments
- fix https://github.com/grpc/grpc/issues/16533 by adding visual-studio-generated GetHashCode and Equals.

### CODE REVIEW

**Comment by jtattermusch** [comment]
FTR: #15423 where the documentation turned out to be inaccurate.

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,964,792      Total (=)      1,964,792

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,887,066      Total (<)     10,887,071

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,964,792      Total (=)      1,964,792

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,887,064      Total (>)     10,887,063

 No significant differences in binary sizes


```

File: src/csharp/Grpc.Core/ChannelOptions.cs [hunk 1]
```diff
@@ -119,10 +121,60 @@ public string StringValue
...
+            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(stringValue);
+            return hashCode;
+        }
+
+        /// <summary>
+        /// Equality operator.
+        /// </summary>
+        public static bool operator ==(ChannelOption option1, ChannelOption option2)
+        {
+            return EqualityComparer<ChannelOption>.Default.Equals(option1, option2);
```
Comment (apolcyn) [inline src/csharp/Grpc.Core/ChannelOptions.cs]: does there need to be an `option1` null check here?

Comment (jtattermusch) [reply src/csharp/Grpc.Core/ChannelOptions.cs]: Yes, I think so, as one can do   `if (null == myChannelOption)` in the code.
This code is actually autogenerated by visual studio, so I have a good confidence that it is correct.

**Pull request merged**